codec-http2/src/main/java/io/netty/handler/codec/http2/internal/hpack/HpackUtil.java

```diff
@@ -31,19 +31,38 @@
  */
 package io.netty.handler.codec.http2.internal.hpack;
 
+import io.netty.util.AsciiString;
+import io.netty.util.internal.ConstantTimeUtils;
+import io.netty.util.internal.PlatformDependent;
+
 final class HpackUtil {
     /**
-     * A string compare that doesn't leak timing information.
+     * Compare two {@link CharSequence} objects without leaking timing information.
+     * <p>
+     * The {@code int} return type is intentional and is designed to allow cascading of constant time operations:
+     * <pre>
+     *     String s1 = "foo";
+     *     String s2 = "foo";
+     *     String s3 = "foo";
+     *     String s4 = "goo";
+     *     boolean equals = (equalsConstantTime(s1, s2) & equalsConstantTime(s3, s4)) != 0;
+     * </pre>
+     * @param s1 the first value.
+     * @param s2 the second value.
+     * @return {@code 0} if not equal. {@code 1} if equal.
      */
-    static boolean equals(CharSequence s1, CharSequence s2) {
-        if (s1.length() != s2.length()) {
-            return false;
-        }
-        char c = 0;
-        for (int i = 0; i < s1.length(); i++) {
-            c |= s1.charAt(i) ^ s2.charAt(i);
+    static int equalsConstantTime(CharSequence s1, CharSequence s2) {
+        if (s1 instanceof AsciiString && s2 instanceof AsciiString) {
+            if (s1.length() != s2.length()) {
```

Comment:
Can you clarify your suggestion? `ConstantTimeUtils` for `CharSequence` does have this check. However the variant which operates on `byte[]` accepts a length parameter ... so it is implied the length which will be iterated is the same (and bounds checks are omitted are assumed to be outside the scope of the method per the docs).